src/api/app/models/cloud/upload_job.rb

```diff
@@ -4,21 +4,55 @@ class UploadJob
     include ActiveModel::Model
     extend Forwardable
 
-    attr_accessor :user_upload_job, :backend_upload_job
-    validate :validate_jobs
+    attr_accessor :user_upload_job, :backend_upload_job, :params
+    validate :validate_jobs, :validate_filename, :validate_architecture, :validate_regions, :validate_virtualization_types, :validate_ami_name
     def_delegator :backend_upload_job, :id
 
     def self.create(user, params)
-      upload_job = new
+      upload_job = new(params: params)
```

Comment:
Just curious, why don't you use super here?

Reply:
Would it make a difference? I just left it like it is ...

Reply:
Right. As said, I was just curious. I shouldn't worry about this.

Reply:
> why don't you use super here?
 
@bgeuken I think because this is not an ActiveRecord model so there is no super create method to call here.